compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -2414,14 +2414,6 @@ private void createSimpleVarDefStmt(BLangSimpleVariable simpleVariable, BLangBlo
 
     @Override
     public void visit(BLangAssignment assignNode) {
-        if (safeNavigateLHS(assignNode.varRef)) {
```

Comment:
Removed this check since this is validated at TypeChecker and logged the errors.